### PR TITLE
sensor create and edit

### PR DESCRIPTION
**Pull request by ps-eqsdigital**
create and edit sensor
edit only available if the user updating is the same as sensor owner

### CODE REVIEW

**Review by js-eqsdigital — APPROVED** [review]

File: api/api/Middleware/TokenValidationMiddleware.cs [hunk 1]
```diff
@@ -24,21 +26,29 @@ public async Task Invoke(HttpContext context)
...
                     if (authorizeAttribute == null)
                     {
-                        await _next(context); 
+                        await _next(context);
                     }
                     else if (!string.IsNullOrWhiteSpace(token))
                     {
-                        var userToken = await userDataAccessObject.GetTokenUuidByToken(token.Substring("Bearer ".Length));
-
+                        var userToken = await userDataAccessObject.GetToken(token.Substring("Bearer ".Length));
```
Comment (kripton69) [inline api/api/Middleware/TokenValidationMiddleware.cs]: types

File: api/api/Controllers/SensorController.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        [HttpPost("insert"), Authorize]
+        public async Task<ActionResult<Guid>> InsertSensor([FromBody] SensorRequest sensor)
+        {
+            OperationResult result = await _sensorBusinessObject.CreateSensor(sensor.ToSensor());
+            if (result.Exception is Exception)
+            {
+                return StatusCode(400);
+            }
+            return Ok(result);
+        }
```
Comment (kripton69) [inline api/api/Controllers/SensorController.cs:32]: break line

File: api/api/Middleware/TokenValidationMiddleware.cs (hunk 1 above)
Comment (ps-eqsdigital) [reply api/api/Middleware/TokenValidationMiddleware.cs]: done

File: api/api/Controllers/SensorController.cs (hunk 2 above)
Comment (ps-eqsdigital) [reply api/api/Controllers/SensorController.cs:32]: done

**Review by kripton69 — APPROVED** [review]

**Pull request merged**